packages/react-google-maps-api/src/components/directions/DirectionsRenderer.tsx

```diff
@@ -8,49 +8,52 @@ import {
 
 import MapContext from '../../map-context'
 
-import { DirectionsRendererPropTypes } from '../../proptypes'
-
 const eventMap = {
   onDirectionsChanged: 'directions_changed',
 }
 
 const updaterMap = {
-  directions (instance, directions) {
+  directions (instance: google.maps.DirectionsRenderer, directions: google.maps.DirectionsResult) {
     instance.setDirections(directions)
   },
-  map (instance, map) {
+  map (instance: google.maps.DirectionsRenderer, map: google.maps.Map) {
     instance.setMap(map)
   },
-  options (instance, options) {
+  options (instance: google.maps.DirectionsRenderer, options: google.maps.DirectionsRendererOptions) {
     instance.setOptions(options)
   },
-  panel (instance, panel) {
+  panel (instance: google.maps.DirectionsRenderer, panel: Element) {
     instance.setPanel(panel)
   },
-  routeIndex (instance, routeIndex) {
+  routeIndex (instance: google.maps.DirectionsRenderer, routeIndex: number) {
     instance.setRouteIndex(routeIndex)
   },
 }
 
-export class DirectionsRenderer extends PureComponent {
-  static propTypes = DirectionsRendererPropTypes
+interface DirectionsRendererState {
+  directionsRenderer?: google.maps.DirectionsRenderer
+}
+
+interface DirectionsRendererProps {
+  options?: google.maps.DirectionsRendererOptions;
+  directions?: google.maps.DirectionsResult;
+  panel?: Element;
+  routeIndex?: number;
+  onDirectionsChanged?: () => void;
+}
 
+export class DirectionsRenderer extends PureComponent<DirectionsRendererProps, DirectionsRendererState> {
   static contextType = MapContext
 
-  registeredEvents = []
+  registeredEvents: google.maps.MapsEventListener[] = []
 
-  state = {
+  state: DirectionsRendererState = {
     directionsRenderer: null
   }
 
   componentDidMount = () => {
     const directionsRenderer = new google.maps.DirectionsRenderer(
-      Object.assign(
-        {
-          map: this.context
-        },
-        this.props.options
-      )
+      this.props.options
```

Comment:
Have you tested this change? my project heavily depends on this component. I do not really want to pass map to options object from outside. I do remember the reason I've written this line of code :)